components/builder-depot/src/server.rs

```diff
@@ -154,10 +157,26 @@ pub fn invite_to_origin(req: &mut Request) -> IronResult<Response> {
     // Lookup the users account_id
     match conn.route::<AccountGet, Account>(&request) {
         Ok(mut account) => {
+            debug!("Successfully found account for user {}, with account ID {}",
+                   &user_to_invite,
+                   &account.get_id());
             invite_request.set_account_id(account.get_id());
             invite_request.set_account_name(account.take_name());
         }
-        Err(err) => return Ok(render_net_error(&err)),
+        Err(err) => {
+            debug!("Failed to find account for user {}. Error = {:?}",
+                   &user_to_invite,
+                   &err);
+
+            if gh.other_user(&session.get_token(), &user_to_invite).is_ok() {
+                return Ok(Response::with((status::NotFound,
+                                          "This is a valid GitHub user but they have not logged \
```

Comment:
I don't think we should put the GitHub user check here on the server but rather inside the client (the web-app). This longer form message is a good hint that we're putting it in the wrong place - the server ideally sends down a useful error code that the client can then localize (if we had other locales). Having this call here will also extend the duration of a failed request on the server.

What I'd recommend is to instead issue two promises inside the web-app when making calls to endpoints that expect a user in the system. Have the first promise be the call to our API and the second promise be the call to GitHub that you're doing here. If the first promise fails with a user not found case then inspect the results of the second promise and see if that got a 200. If it did, then we can show this error message to the user.